velox/functions/prestosql/registration/GeneralFunctionsRegistration.cpp

```diff
@@ -17,9 +17,36 @@
 #include "velox/functions/Registerer.h"
 #include "velox/functions/lib/IsNull.h"
 #include "velox/functions/prestosql/Cardinality.h"
+#include "velox/functions/prestosql/GreatestLeast.h"
 #include "velox/functions/prestosql/InPredicate.h"
 
 namespace facebook::velox::functions {
+
+template <typename T>
+inline void registerGreatestLeastFunction(const std::string& prefix) {
+  registerFunction<ParameterBinder<GreatestFunction, T>, T, T, Variadic<T>>(
+      {prefix + "greatest"});
+
+  registerFunction<ParameterBinder<LeastFunction, T>, T, T, Variadic<T>>(
+      {prefix + "least"});
+}
+
+inline void registerAllGreatestLeastFunctions(const std::string& prefix) {
+  registerGreatestLeastFunction<bool>(prefix);
+  registerGreatestLeastFunction<int8_t>(prefix);
+  registerGreatestLeastFunction<int16_t>(prefix);
+  registerGreatestLeastFunction<int32_t>(prefix);
+  registerGreatestLeastFunction<int64_t>(prefix);
+  registerGreatestLeastFunction<int128_t>(prefix);
```

Comment:
This should be removed. int128_t is used for LONG_DECIMAL type, which you are registering on L44.

I'll fix this, but wanted to let you know.

Reply:
@mbasmanova 
![image](https://github.com/facebookincubator/velox/assets/54862932/6d3101da-183a-4647-b574-70f5dce9b93c)
![image](https://github.com/facebookincubator/velox/assets/54862932/74d07082-0b76-444c-89fc-c12d0ad8aaf7)
According to type definitions, it seems like ShortDecimal is using int64_t, and the ShortDecimal is also being registered. I am curious in this case do we need to remove `registerGreatestLeastFunction<int64_t>(prefix);` in addition?

Reply:
@Real-Chen-Happy Function registry stores a mapping from function signature to function implementation. Function signature is defined using logical type, e.g. f(integer) is different from f(date) even though both integer and date types are backed by INTEGER. Hence, we need a register call for each signature we wish to support. Since we want to support least(bigint,...), we need to call `registerFunction<int64_t>`. 

Now, there is only one type backed by int128_t, that is long decimal. It actually seems like a bug that ``registerFunction<int128_t>` call is allowed. I think it should not be allowed.

Reply:
@mbasmanova I am thinking `registerFunction<int128_t>` may be reserved for HUGEINT?

Also, it seems like we are using logical types (ShortDecimal ...), Physical types (Varchar), and C++ types (int8_t ...) in function registrations, which is a bit confusing. Is there any future plan where we can just use physical types and logical types for function registrations?

Reply:
There is actually no HUGEINT type in Velox. There is TypeKind::HUGEINT though. Things could definitely be clearer. Too many things to fix / improve. Not enough time.